sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -59,6 +58,9 @@ object PartitionSpec {
 
 object PartitioningUtils {
 
+  val datePartitionPattern = "yyyy-MM-dd"
```

Comment:
looks like we can create the `dateFormatter` here.

Reply:
I think to remove this pattern for date at all because it equals to default date pattern for `DateFormatter`.